packages/editor/src/components/post-status/index.js

```diff
@@ -0,0 +1,318 @@
+		postType,
+		isBeingScheduled,
+		canEdit,
+	} = useSelect( ( select ) => {
+		const {
+			getEditedPostAttribute,
+			getCurrentPostId,
+			getCurrentPostType,
+			isEditedPostBeingScheduled,
+			getCurrentPost,
+		} = select( editorStore );
+		// const post = select( editorStore ).getCurrentPost();
+		return {
+			status: getEditedPostAttribute( 'status' ),
+			date: getEditedPostAttribute( 'date' ),
+			password: getEditedPostAttribute( 'password' ),
+			postId: getCurrentPostId(),
+			postType: getCurrentPostType(),
+			isBeingScheduled: isEditedPostBeingScheduled(),
+			canEdit: getCurrentPost()._links?.[ 'wp:action-publish' ] ?? false,
+		};
+	}, [] );
+	const [ showPassword, setShowPassword ] = useState( !! password );
+	const passwordInputId = useInstanceId(
+		PostStatus,
+		'editor-change-status__password-input'
+	);
+	const { editEntityRecord } = useDispatch( coreStore );
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const [ popoverAnchor, setPopoverAnchor ] = useState( null );
+	// Memoize popoverProps to avoid returning a new object every time.
+	const popoverProps = useMemo(
+		() => ( {
+			// Anchor the popover to the middle of the entire row so that it doesn't
+			// move around when the label changes.
+			anchor: popoverAnchor,
+			'aria-label': __( 'Status & visibility' ),
+			headerTitle: __( 'Status & visibility' ),
+			placement: 'left-start',
+			offset: 36,
+			shift: true,
+		} ),
+		[ popoverAnchor ]
+	);
+
+	// Effect to update post status when post date changes.
+	useEffect( () => {
+		if (
+			isBeingScheduled &&
+			! [ 'future', 'private', 'pending' ].includes( status )
+		) {
+			editEntityRecord( 'postType', postType, postId, {
+				status: 'future',
+			} );
+		}
+	}, [ isBeingScheduled, editEntityRecord, status, postType, postId ] );
+
+	const updatePost = async ( {
+		status: newStatus = status,
+		password: newPassword = password,
+		date: newDate = date,
+	} ) => {
+		try {
+			await editEntityRecord( 'postType', postType, postId, {
+				status: newStatus,
+				date: newDate,
+				password: newPassword,
+			} );
+		} catch ( error ) {
+			const errorMessage =
+				error.message && error.code !== 'unknown_error'
+					? error.message
+					: __( 'An error occurred while updating the status' );
+
+			createErrorNotice( errorMessage, {
+				type: 'snackbar',
+			} );
+		}
+	};
+
+	const handleTogglePassword = ( value ) => {
+		setShowPassword( value );
+		if ( ! value ) {
+			updatePost( { password: '' } );
+		}
+	};
+
+	const handleStatus = ( value ) => {
+		let newDate = date;
+		let newPassword = password;
+		if ( [ 'publish', 'draft' ].includes( value ) ) {
+			if ( new Date( date ) > new Date() ) {
+				newDate = new Date();
+			}
+		} else if ( value === 'future' ) {
+			if ( ! date || new Date( date ) < new Date() ) {
+				newDate = new Date();
+				newDate.setDate( newDate.getDate() + 7 );
+			}
+		} else if ( value === 'private' && password ) {
+			setShowPassword( false );
```

Comment:
I don't think this call is needed. we already hide the password toggle if the status is private.